runtime/src/accounts_db.rs

```diff
@@ -1608,7 +1609,9 @@ impl AccountsDb {
                                     }
                                 }
                                 AccountIndexGetResult::NotFoundOnFork => {
-                                    // do nothing - pubkey is in index, but not found in a root slot
+                                    // pubkey is in the index but not in a root slot, so clean it
```

Comment:
An additional clarifying comment is that also this pubkey must have been touched by some slot since it was in the dirty list, so we assume that the slot it was touched in must be unrooted

Reply:
Done!